File: CodeChef/picking elements.py
temp=input()
temp=temp.split(" ")
n=int(temp[0])
m=int(temp[1])
arr=[]
temp=input()
temp=temp.split(" ")
for i in range(n):
	arr.append(int(temp[i]))

divisor=[]
for i in range(100005):
	divisor.append(0)

for i in range(1,100002):
	j=1
	while (i*j<100002):
		divisor[i*j]+=1

arr.sort()
arr2=[]
arr2.append(arr[0])
div=[]
div2=[]
div.append(divisor[arr[0]])
div2.append(divisor[arr[0]])
for i in range(1,n):
	div.append(divisor[arr[i]])
	div2.append(div2[i-1]+divisor[arr[i]])
	arr2.append([arr2[i-1]+arr[i]])
cost=0
count=0
cost=arr[0]+divisor[0]*1
while (cost<=m):
	count+=1
	cost=arr[count]+div2[count]*(count+1)
print count

File: picking_elements.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){

	#ifndef ONLINE_JUDGE
    // for getting input from input.txt
    freopen("input.txt", "r", stdin);
    // for writing output to output.txt
    freopen("output.txt", "w", stdout);
	#endif

    int n,m;
    cin >> n >> m;
    int arr[n];
    int i;
    for (i=0;i<n;i++){
    	cin >> arr[i];
    }
    int divisor[100005]={0};
    int j;
    for (i=1;i<100002;i++){
    	for (j=1;i*j<100002;j++){
    		divisor[i*j]++;
    	}
    }
    sort(arr,arr+n);
    int divisor2[100005]={0},arr2[n];
    arr2[0]=arr[0];
    for (i=1;i<n;i++){
    	arr2[i]=arr2[i-1]+arr[i];
    }
    for (i=1;i<100002;i++){
    	divisor2[i]=divisor2[i-1]+divisor[i];
    }



	
	return 0;
}